### PR TITLE
Remove inventory unit state

### PR DESCRIPTION
**Pull request by mbj**
Depends on #314

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Ready for review. If you wanted I could merge this into #314.

**Comment by dkubb** [comment]
Has this been rebased on top of the #314 branch? As far as I know it's specs are failing now.

**Comment by mbj** [comment]
@dkubb No, I'll fix #314 before I touch this one again.

**Comment by mbj** [comment]
@dkubb Assuming all green this is ready for review.

File: core/spec/models/spree/shipment_spec.rb [hunk 1]
```diff
@@ -134,17 +134,17 @@ def create_shipment(order, stock_location)
...
         before do
           allow(shipment).to receive(:inventory_units) { inventory_units }
           allow(inventory_units).to receive_message_chain(:includes, :joins).and_return inventory_units
         end
 
-        it 'should use symbols for states when adding contents to package' do
+        it 'should reflect quantity in package' do
           package = shipment.to_package
-          expect(package.on_hand.count).to eq 1
+          expect(package.quantity).to eq 1
```
Comment (dkubb) [inline core/spec/models/spree/shipment_spec.rb]: `be(1)`

File: core/spec/models/spree/stock/package_spec.rb [hunk 2]
```diff
@@ -16,28 +16,20 @@ def build_inventory_unit
...
-        item = subject.find_item(unit, :on_hand)
-        expect(item).to be_nil
+        expect(subject.find_item(build_inventory_unit)).to be(nil)
       end
 
       it 'finds content item for an inventory unit' do
         unit = build_inventory_unit
         subject.add unit
-        item = subject.find_item(unit, :on_hand)
+        item = subject.find_item(unit)
```
Comment (dkubb) [inline core/spec/models/spree/stock/package_spec.rb]: I don't think it's necessary to assign to the `item` synthetic variable and then use it on the line below.

I would change the line below to:

```ruby
expect(subject.find_item(unit).quantity).to be(1)
```

File: core/spec/models/spree/stock/packer_spec.rb [hunk 3]
```diff
@@ -24,7 +24,7 @@ module Stock
           expect(stock_location).to receive(:fill_status).exactly(5).times.and_return(3, 1, 2, 0)
 
           package = subject.default_package
-          expect(package.on_hand.size).to eq 3
+          expect(package.quantity).to eq 3
```
Comment (dkubb) [inline core/spec/models/spree/stock/packer_spec.rb]: `be(3)`

File: core/spec/models/spree/order_inventory_spec.rb [hunk 4]
```diff
@@ -37,8 +37,6 @@
     context 'when more inventory units exist than quantity' do
       before do
         line_item.inventory_units.create!(
-          order:    order,
```
Comment (dkubb) [inline core/spec/models/spree/order_inventory_spec.rb]: Does the `order_id` column still become set correctly when this is explicit assignment is removed?

File: core/spec/models/spree/shipment_spec.rb [hunk 5]
```diff
@@ -134,17 +134,17 @@ def create_shipment(order, stock_location)
 
       context 'to_package' do
         let(:inventory_units) do
-          [build(:inventory_unit, state: 'on_hand')]
+          [build(:inventory_unit)]
```
Comment (dkubb) [inline core/spec/models/spree/shipment_spec.rb]: You can probably change this to a one liner, eg:

```ruby
let(:inventory_units) { [build(:inventory_unit)] }
```

File: core/app/models/spree/inventory_unit.rb [hunk 6]
```diff
@@ -5,28 +5,11 @@ class InventoryUnit < Spree::Base
     belongs_to :shipment, class_name: "Spree::Shipment", touch: true, inverse_of: :inventory_units
     belongs_to :line_item, class_name: "Spree::LineItem", inverse_of: :inventory_units
 
-    scope :on_hand, -> { where state: 'on_hand' }
-
-    state_machine initial: :on_hand
-
-    validates :state, :variant, :order, :shipment, :line_item, presence: true
-    validates :state,   inclusion: { in: %w[on_hand] }
-    validates :pending, inclusion: { in: [true, false] }
```
Comment (dkubb) [inline core/app/models/spree/inventory_unit.rb]: Did you mean to remove the validation for the `pending` field?

File: core/spec/models/spree/order_inventory_spec.rb (hunk 4 above)
Comment (mbj) [inline core/spec/models/spree/order_inventory_spec.rb]: No, this is an untested edit, where I meant only to remove the `state` below. Just turned CI red. good catch.

File: core/app/models/spree/stock/package.rb [hunk 7]
```diff
@@ -33,20 +33,14 @@ def weight
...
-      def on_hand
-        contents.select(&:on_hand?)
-      end
-
-      def find_item(inventory_unit, state = nil)
+      def find_item(inventory_unit)
         contents.detect do |item|
-          item.inventory_unit == inventory_unit &&
-            (!state || item.state.to_s == state.to_s)
+          item.inventory_unit.eql?(inventory_unit)
```
Comment (dkubb) [inline core/app/models/spree/stock/package.rb]: Can this whole loop be transformed into a one-liner without exceeding 79 chars?

File: core/app/models/spree/inventory_unit.rb (hunk 6 above)
Comment (mbj) [inline core/app/models/spree/inventory_unit.rb]: Jup, as there is no attribute in the DB. Its a leftover.

Comment (dkubb) [inline core/app/models/spree/inventory_unit.rb]: I still see the `pending` attribute in the spree db structure: https://github.com/MountainRoseHerbs/spree/pull/326/files#diff-0d962d77904d8cdc45a07a9aea97325cR531

Comment (mbj) [inline core/app/models/spree/inventory_unit.rb]: I'm blind. It'll get removed next commit anyway, I'll restore the pending validation meanwhile.

**Pull request merged**